.github/workflows/lockfiles.yaml

```diff
@@ -0,0 +1,409 @@
+
+      - name: Create branch for new PR
+        if: ${{ inputs.pr == 'new' }}
+        run: |
+          git config --local user.name "github-actions[bot]"
+          git config --local user.email "41898282+github-actions[bot]@users.noreply.github.com"
+          git checkout -b "${{ needs.pr.outputs.PR_REF }}" FETCH_HEAD
+
+      - name: Download lockfiles and lockfile diff files
+        uses: actions/download-artifact@v5
+        with:
+          pattern: lockfile-* # lockfile-{resolve}
+          path: ${{ env.OUTPUT_BASE_DIR }}
+          merge-multiple: true # unpack in {path}/{artifact_name}/
+
+      - name: Prepare commit
+        run: |
+          echo "pants generate-lockfiles: ${RESOLVES_CSV}" > ${COMMIT_MSG}
+          echo >> ${COMMIT_MSG}
+          for resolve in ${RESOLVES}; do
+            LOCKFILE=$(jq '.["'"${resolve}"'"]' <<< "${LOCKFILES}")
+            cp "${{ env.OUTPUT_BASE_DIR }}/lockfile-${resolve}/$(basename ${LOCKFILE})" "${LOCKFILE}"
+            git add "${LOCKFILE}"
+            STDERR_LOG="${{ env.OUTPUT_BASE_DIR }}/lockfile-${resolve}/stderr.log"
+            if [ -e ${STDERR_LOG}.txt ]; then
+              cat ${STDERR_LOG}.txt >> ${COMMIT_MSG}
+            else
+              echo "${STDERR_LOG}.txt is missing"
+              echo "No changes to: ${LOCKFILE}" >> ${COMMIT_MSG}
+              echo >> ${COMMIT_MSG}
+            fi
+          done
+
+      - name: Prepare PR comment
+        env:
+          RUN_LINK: ${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }})
+        run: |
+          echo "# Lockfile Diffs" > ${PR_COMMENT}
+          echo >> ${PR_COMMENT}
+          for resolve in ${RESOLVES}; do
+            LOCKFILE=$(jq '.["'"${resolve}"'"]' <<< "${LOCKFILES}")
+            STDERR_LOG="${{ env.OUTPUT_BASE_DIR }}/lockfile-${resolve}/stderr.log"
+            if [ -e ${STDERR_LOG}.md ]; then
+              cat ${STDERR_LOG}.md >> ${PR_COMMENT}
+            else
+              echo "${STDERR_LOG}.md is missing"
+              echo "No changes to: `${LOCKFILE}`" >> ${PR_COMMENT}
+              echo >> ${PR_COMMENT}
+            fi
+          done
+          echo ":robot: [GitHub Actions Workflow Run](${RUN_LINK})" >> ${PR_COMMENT}
+
+      - name: Commit and push
+        run: |
+          git commit -F "${COMMIT_MSG}"
+          git push -u origin "${{ needs.pr.outputs.PR_REF }}"
+
+      - name: Create new PR
+        if: ${{ inputs.pr == 'new' }}
+        run: >
+          gh pr create
+          --base "${{ needs.pr.outputs.PR_BASE_REF }}"
+          --title "pants generate-lockfiles: ${RESOLVES_CSV}"
+          --body-file "${PR_COMMENT}"
+          --reviewer "Maintainers"
+          --assignee "${{ github.event.sender.login }}"
+          --label "external dependency"
+          --label "python3"
+
+      # TODO This updates an existing comment. Should it just add a new comment on re-run?
```

Comment:
To add a new comment, we would remove `--edit-last --create-if-none` from `run`. I'm leaning towards just deleting this TODO. Most bots seem to update their comment instead of creating multiple.